components/provisioner/pkg/gqlschema/schema.graphql

```diff
@@ -150,6 +151,7 @@ input GardenerConfigInput {                   # Gardener project in which the cl
     machineType: String!                            # Type of node machines, varies depending on the target provider
     region: String!                                 # Region in which the cluster is created
     provider: String!                               # Target provider on which to provision the cluster (Azure, AWS, GCP)
+    purpose: String                                 # Purpose is the purpose class for this cluster
```

Comment:
Just a question:  I'm not sure, why that this field is nullable. Do we assume KEB can send us the configuration without setting the purpose field?

Reply:
KEB should always send this field. This is GraphQL though and the Gardener has a default value for it so I think it's good to leave it optional. This approach also would not provide any problems with synchronization because both calls - with and without the `purpose` field will work